node/resolvers/sponsoredProducts/newtail.ts

```diff
@@ -114,13 +118,37 @@ export async function newtailSponsoredProducts(
   try {
     const adsAmount = args.sponsoredCount ?? DEFAULT_SPONSORED_COUNT
 
-    const categoryName =
-      args?.selectedFacets?.length && args.selectedFacets.some((facet) => (facet.key.startsWith("category") || facet.key === "c" || facet.key === "categoria"))
-        ? args.selectedFacets
-            .filter((facet) => (facet.key.startsWith("category") || facet.key === "c" || facet.key === "categoria"))
-            .map((facet) => facet.value)
-            .join(" > ")
-        : undefined;
+    const categoryFacets = args?.selectedFacets?.filter(
+      (facet) => CATEGORY_FACET_KEYS.includes(facet.key?.toLowerCase())
+    ) || []
+
+    let categoryName: string | undefined
```

Comment:
This logic could be extracted into a separate method or class. Including it here makes it difficult to understand the main purpose of this method, which is to retrieve sponsored products.

Reply:
What do you think?

Reply:
I totally agree! will work on that if we keep this PR